R/slider.R

```diff
@@ -0,0 +1,48 @@
+#' Slider Module
+#' 
+#' @description Generates a slider which can animate from minimum to maximum value.
+#'
+#' @param id An ID string that corresponds with the ID used to call the module's
+#'   UI function
+#' @param label Character string with which the slider will be labelled. Default is NULL - no label.
+#' @param min The minimum value that can be selected
+#' @param max The maximum value that can be selected
+#' @param value The initial value of the slider
+#' @param interval Number specifying how fast animation should progress
+#' @param loop Should the animation be a loop. Default is FALSE - no loop.
+#' 
+#' @author Mateusz Wyszynski
+#' @export
+sliderUI <- function(id, label = NULL, min, max, value, interval, loop = FALSE) {
```

Comment:
We don't want to add new requirements to slider (we don't want to make min a required value for example). The aim of this package is just to make slider easier, i.e. to allow user just write two lines to build a slider and to allow him use namespaces easily. This whole module should consist of following (plus comments):
```
sliderUI <- function(id, ...) {
  ns <- NS(id)

  sliderInput(ns("slider"), ...)
}

slider <- function(input, output, session) {
  return(reactive(input$slider))
}
```

Then user can create slider by calling:
```
sliderUI("damian_slider", "Label", 1, 100, 1, 10, animate = animationOptions(interval = 10, loop = TRUE)
```
and on the server side he would just call:
```
slider_value <- callModule(slider, "damian_slider")
```

Please change the code accordingly.